src/solareclipseworkbench/reference_moments.py

```diff
@@ -89,20 +123,59 @@ def calculate_reference_moments(location: EarthLocation, time: Time) -> dict:
     # Calculate the start/end points of partial/total solar eclipse
     partial = np.flatnonzero(amount > 0)
     timings = {}
+    alt, az = __calculate_alt_az(ts, earth, sunc, loc, sunrise.utc_datetime()[0])
+    sunrise = ReferenceMomentInfo(sunrise.utc_datetime()[0], az, alt)
+    timings['rise'] = sunrise
+
     if len(partial) > 0:
         start_partial, end_partial = times[partial[[0, -1]]]
-        timings["C1"] = start_partial.isot
+        alt, az = __calculate_alt_az(ts, earth, sunc, loc, start_partial.datetime)
+        c1 = ReferenceMomentInfo(start_partial.datetime, az, alt)
+        timings["C1"] = c1
 
         total = np.flatnonzero(amount_minimum == 1)
         if len(total) > 0:
             start_total, end_total = times[total[[0, -1]]]
-            timings["C2"] = start_total.isot
-            timings["MAX"] = Time((start_total.unix + end_total.unix) / 2, format="unix").isot
-            timings["C3"] = end_total.isot
-        timings["C4"] = end_partial.isot
+            alt, az = __calculate_alt_az(ts, earth, sunc, loc, start_total.datetime)
+            c2 = ReferenceMomentInfo(start_total.datetime, az, alt)
+            timings["C2"] = c2
+
+            max_time = Time((start_total.unix + end_total.unix) / 2, format="unix").datetime
+            alt, az = __calculate_alt_az(ts, earth, sunc, loc, max_time)
+            max = ReferenceMomentInfo(max_time, az, alt)
+            timings["MAX"] = max
+
+            alt, az = __calculate_alt_az(ts, earth, sunc, loc, end_total.datetime)
+            c3 = ReferenceMomentInfo(end_total.datetime, az, alt)
+            timings["C3"] = c3
+
+            timings["duration"] = (end_total - start_total).datetime
+            max_time = (start_total.unix + end_total.unix) / 2
+        else:
+            max_time = Time((start_partial.unix + end_partial.unix) / 2, format="unix").datetime
+            alt, az = __calculate_alt_az(ts, earth, sunc, loc, max_time)
+            max = ReferenceMomentInfo(max_time, az, alt)
+            timings["MAX"] = max
+        max_loc = location.get_itrs(Time(max_time, format="unix"))
+        timings["magnitude"] = sun.eclipse_amount(max_loc).value / 100
```

Comment:
I would not add the magnitude to the `timings` dictionary, but instead return the `timings` dictionary and the magnitude separately.